### PR TITLE
[wasm] Remove dead code and add fetch callback to mono_load_runtime_and_bcl.

### PR DESCRIPTION
**Pull request by kumpera**
The mono_load_runtime_and_bcl allows upstream to remap requests and report download progress.

This cb semantics might change in the future when we implement more agressive loading.

### CODE REVIEW

**Comment by kumpera** [comment]
CC @jeromelaban @SteveSandersonMS

File: sdks/wasm/library_mono.js [hunk 1]
```diff
@@ -72,14 +72,20 @@ var MonoSupportLib = {
...
 
-		mono_load_runtime_and_bcl: function (vfs_prefix, deploy_prefix, enable_debugging, file_list, loaded_cb) {
+		mono_load_runtime_and_bcl: function (vfs_prefix, deploy_prefix, enable_debugging, file_list, loaded_cb, fetch_file_cb) {
 			Module.FS_createPath ("/", vfs_prefix, true, true);
 
 			var pending = 0;
 			var loaded_files = [];
 			file_list.forEach (function(file_name) {
 				++pending;
-				fetch (deploy_prefix + "/" + file_name, { credentials: 'same-origin' }).then (function (response) {
```
[the remark this answers is not in the document]
Comment (kumpera) [reply sdks/wasm/library_mono.js]: That's the diff line removing that line, it was moved to line 86

[the remark this answers is not in the document]
Comment (jeromelaban) [reply sdks/wasm/library_mono.js]: Was actually for the next line, about `fetch_promise` being `fetch_promisse`.

**Comment by lewurm** [comment]
@monojenkins build failed

**Comment by lewurm** [comment]
@monojenkins squash

**Pull request merged**